src/algebra/monoid_algebra.lean

```diff
@@ -82,13 +86,88 @@ instance : non_unital_non_assoc_semiring (monoid_algebra k G) :=
   add           := (+),
   left_distrib  := assume f g h, by simp only [mul_def, sum_add_index, mul_add, mul_zero,
     single_zero, single_add, eq_self_iff_true, forall_true_iff, forall_3_true_iff, sum_add],
-  right_distrib := assume f g h, by simp only [mul_def, sum_add_index, add_mul, mul_zero, zero_mul,
+  right_distrib := assume f g h, by simp only [mul_def, sum_add_index, add_mul, zero_mul,
     single_zero, single_add, eq_self_iff_true, forall_true_iff, forall_3_true_iff, sum_zero,
     sum_add],
   zero_mul  := assume f, by simp only [mul_def, sum_zero_index],
   mul_zero  := assume f, by simp only [mul_def, sum_zero_index, sum_zero],
   .. finsupp.add_comm_monoid }
 
+instance is_scalar_tower_self : is_scalar_tower k (monoid_algebra k G) (monoid_algebra k G) :=
+⟨λ t a b, show (t • a) * b = t • (a * b),
```

Comment:
Are you sure this isn't a less general case of `is_scalar_tower` below?

Reply:
Ah, I'm thinking of a generalized multiplication we don't have yet

Reply:
I believe this is a different instance because it makes a statement about the convolution multiplication `monoid_algebra.has_mul` whereas the instance [here](https://github.com/leanprover-community/mathlib/blob/461b444d065e1747f1bda60070b4b5d6a3026fb2/src/algebra/monoid_algebra.lean#L240) is a statement about the pointwise multiplication defined in `finsupp`.

Note also that to apply the instance below, we would need a `monoid` structure on `monoid_algebra k G` which we do not have.